### PR TITLE
Expand try/except block to ensure that failures in the pipeline get handled properly and instances are stopped/terminated appropriately

### PR DESCRIPTION
**Pull request by jcasner**
@lscheinkman @unixorn please review

### CODE REVIEW

File: grok-mobile/run_pipeline [hunk 1]
```diff
@@ -298,31 +298,32 @@ def main(args):
...
+                 os.path.join(GROK_MOBILE_HOME, "android"))
+
+    artifactsDir = setupArtifactsDir()
+
+    # Build mobile client
+    with changeToWorkingDir(ANDROID_PATH):
+      runWithOutput("./gradlew clean build %s %s" %
+                      (g_googleAnalyticsIdParameter,
+                       g_feedbackEmailParameter),
+                    env=os.environ, logger=logger)
```
Comment (unixorn) [inline grok-mobile/run_pipeline]: indentation

**Comment by unixorn** [comment]
LGTM after fixing the indentation

File: grok-mobile/run_pipeline (hunk 1 above)
Comment (jcasner) [inline grok-mobile/run_pipeline]: What's wrong with it?  It looks OK to me...

Comment (unixorn) [inline grok-mobile/run_pipeline]: Actually, you were right, I was wrong, leave it

**Pull request merged**